RobotsAndPencils/slack-salesforce-bot
# Salesforce rest client

#### What's this PR do?
Adds the ability to save notes in salesforce from bot commands
#### Where should the reviewer start?
`services/salesforce`
#### How should this be manually tested?
```
source setenv.sh
gb build all; ./bin/bot
```
go to slack and say something like `note we’d actually like 200 lbs of paper 100 lbs is just not gonna be enough` Look in salesforce you'll see a new note https://cs14.salesforce.com/006c000000EMa77AAD

#### Any background context you want to provide?
I wasn't sure how we'd actually like to gather the text and persist the note. A direct bot command probably isn't the way to go but wanted to get the plumbing in place to persist notes in salesforce. When we determine how we'd like to gather the note text and title `salesforce.CreateNote` should do the trick  getting it there. We also need to determine how we want to get the opportunity ID to set as `ParentID` on the note struct for now it's hard coded.
#### Screenshots (if appropriate)
<img width="747" alt="screen shot 2016-02-21 at 11 30 02 am" src="https://cloud.githubusercontent.com/assets/270735/13205580/3f6eff30-d88f-11e5-989a-314ea594e20d.png">

#### Reviewer notes
@randompi  Fell free to make changes as you see fit, I'll be off line for the coming week.
